processing/src/main/java/org/apache/druid/segment/virtual/ExpressionPlan.java

```diff
@@ -277,6 +277,9 @@ public ColumnCapabilities inferColumnCapabilities(@Nullable ColumnType outputTyp
             // until query time
             return ColumnCapabilitiesImpl.copyOf(underlyingCapabilities)
                                          .setType(ColumnType.STRING)
+                                         // this is sad, but currently dictionary encodedness is tied to string
+                                         // selectors and sad stuff happens if the input type isn't string
+                                         .setDictionaryEncoded(underlyingCapabilities.is(ValueType.STRING))
```

Comment:
>shouldn't this be underlyingCapabilities.isTrue() && underlyingCapabilities.is(ValueType.STRING)?

oops yes.

>also, imo copyOf is not the right thing to use here, logically speaking. I don't think we actually want to carry through all caps from the underlying selector, just a few specific ones.

Ah yea that is reasonable i suppose, i think originally more were preserved than not so i went with the copy. I guess `hasBitmapIndexes` isn't widely used on query side anymore since stuff can just ask `ColumnIndexSupplier` for an index and it will return if it has it, though i suppose it should still be preserved just in case.

Reply:
actually, we were explicitly marking hasBitmapIndexes as false, though #15585 allows expressions to use them in many cases, so it probably should be allowed to pass through.